src/components/settings-menu/NotificationsSection.tsx

```diff
@@ -1,122 +1,401 @@
-                'settings.notifications_section.received'
-              )}, ${lang.t('settings.notifications_section.sold')}, ${lang.t(
-                'settings.notifications_section.minted'
-              )}, ${lang.t('settings.notifications_section.plus_n_more', {
-                n: 4,
-              })}`}
-            />
-          }
-          leftComponent={
-            <ContactAvatar
-              alignSelf="center"
-              color={colors.red}
-              marginRight={10}
-              size="small"
-              value="pugson.eth"
-            />
-          }
-          onPress={() => onPress('pugson.eth')}
-          size={52}
-          titleComponent={<MenuItem.Title text="pugson.eth" />}
-        />
-      </Menu>
-      <Menu>
-        <MenuItem
-          disabled
-          rightComponent={<Switch />}
-          size={52}
-          titleComponent={
-            <MenuItem.Title
-              text={lang.t('settings.notifications_section.watched_wallets')}
-              weight="bold"
-            />
-          }
-        />
-        <MenuItem
-          hasRightArrow
-          labelComponent={
-            <MenuItem.Label
-              text={lang.t('settings.notifications_section.off')}
-            />
-          }
-          leftComponent={
-            <ContactAvatar
-              alignSelf="center"
-              color={colors.red}
-              marginRight={10}
-              size="small"
-              value="moxey.eth"
-            />
-          }
-          onPress={() => onPress('moxey.eth')}
-          size={52}
-          titleComponent={<MenuItem.Title text="moxey.eth" />}
-        />
-      </Menu>
-    </MenuContainer>
+          )}
+        </Box>
+      }
+      onPress={() => navigateToWalletSettings(walletName, wallet.address)}
+      size={52}
+      titleComponent={<MenuItem.Title text={walletName} />}
+    />
+  );
+};
+
+const NotificationsSection = () => {
+  const { navigate } = useNavigation();
+  const { colors } = useTheme();
+  const { chainId } = useAccountSettings();
+  const notMainnet = chainId !== 1;
+  const { wallets } = useWallets();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const walletIDs = Object.keys(wallets!);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const ownedWallets: any[] = useMemo(() => [], []);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const watchedWallets: any[] = useMemo(() => [], []);
+  const [walletsLoaded, setWalletsLoaded] = useState(false);
+  const {
+    ownerEnabled,
+    watcherEnabled,
+    updateGroupSettings,
+  } = useWalletGroupNotificationSettings();
+  const noOwnedWallets = walletsLoaded && !ownedWallets.length;
+  const noWatchedWallets = walletsLoaded && !watchedWallets.length;
+  const [permissionStatus, setPermissionStatus] = useState<string | boolean>(
+    false
+  );
+  const noPermissions = permissionStatus !== RESULTS.GRANTED;
+
+  useEffect(() => {
```

Comment:
I don't understand why can't you just populate those values immediately inside `useMemo` (or `useMemoOne`) and you need to use `useEffect` for this, then change state and effectively cause additional rerender and that seems to be very unnecessary

Reply:
seems to me that `useMemo` blocks the render thread and adds extra latency to the navigation event when user has a lot of wallets in the app in this instance. will have to do a bit more stress testing on a real device — maybe not an issue there.

i refactored the code with `useMemo` so it's cleaner, let's see if anyone notices the latency or if i'm just overly sensitive.